### DISCUSSION TITLE
Use python logging to populate junit-xml without showing up in console

### DISCUSSION THREAD

**Discussion by redlars** [pytest-dev/pytest#8559] [Q&A]
How can logging information from pytest be piped to junit-xml without being shown in console output? Used Windows 10 to test this code.

We have the following simple unittest:

    import logging
    import unittest 
    class fixture(unittest.TestCase):

    def test_log(self):
       logging.basicConfig()
       logger = logging.getLogger(__name__)
       logger.info("Hello from inside a test")

    if __name__ == '__main__':
        unittest.main()

The unittest is invoked as shown below, notice that diagnostics info message is included in console output. This is obviously a simple example but in a scenario with many tests and logging messages the console will quickly become very messy. We would therefore prefer that logging messages are not displayed in console and only show in the junit-xml file. We not be able to configure this setup yet.

    py -m pytest test_1.py -v --capture=fd --junit-xml output.xml
    ========================== test session starts==================================
    platform win32 -- Python 3.9.4, pytest-6.2.3, py-1.10.0, pluggy-0.13.1 -- 
    C:\Users\JohnDoe\AppData\Local\Programs\Python\Python39\python.exe
    cachedir: .pytest_cache
    rootdir: C:\work\pytest_logging, configfile: pytest.ini
    collected 1 item

    test_1.py::fixture::test_log
    ---------------------------------- live log call ----------------------------------------------------------------------
    20:09:03:INFO:Hello from inside a test
    PASSED

    -------------------------------- generated xml file: C:\work\pytest_logging\output.xml ---------------------------
    ======================== 1 passed in 0.03s ===================================   

The pytest.ini file looks like this

    [pytest]
    log_cli = true
    log_cli_level = INFO
    log_cli_format = %(asctime)s:%(levelname)s:%(message)s
    junit_logging=log

The junit-xml file looks like this;
<?xml version="1.0" encoding="utf-8"?>
    <testsuites>
     <testsuite name="pytest" errors="0" failures="0" skipped="0" tests="1" time="0.027" timestamp="2021-04-15T20:09:03.216546" hostname="myhost">
       <testcase classname="test_1.fixture" name="test_log" time="0.001">
        <system-out>--------------------------------- Captured Log ---------------------------------
    #x1B[32mINFO    #x1B[0m test_1:test_1.py:9 Hello from inside a test
     --------------------------------- Captured Out ---------------------------------
        </system-out>
        <system-err>--------------------------------- Captured Err ---------------------------------
        </system-err>
      </testcase>
     </testsuite>
    </testsuites>

A second tiny question, does pytest allow formatting the message that is displayed in junit-xml? The current text is a bit verbose and contains what I am guessing is color values for parts of the text.

**Comment by redlars** [comment]
Appreciate any input on this matter.

**Reply by nicoddemus** [reply]
Hi @redlars,

See if `--show-capture` can help you out:

```
  --show-capture={no,stdout,stderr,log,all}
    Controls how captured stdout/stderr/log is shown on failed tests. Default is 'all'.
```

**Reply by redlars** [reply]
Hi @nicoddemus,

Appreciate your input.

The argument did not control the console output from logging.info("...").
The argument did control writing directly to stdout or stderr (see code below) but only when test failed.

    sys.stdout.write('Write some text\n\n\n')
